mbedtls/src/ssl/context.rs

```diff
@@ -225,6 +225,12 @@ impl<T> Context<T>  {
     // This function ultimately ensure the semitics:
     // Returned value `Ok(n)` always means n bytes of data has been sent into c-mbedtls's buffer (some of them might be sent out through underlying IO)
     pub(super) fn async_write(&mut self, buf: &[u8]) -> Result<usize> {
+        let max_len = unsafe { ssl_get_max_out_record_payload((&*self).into()).into_result()? as usize };
+        #[cfg(feature = "zlib")]
+        static RESERVE_SIZE: usize = 10;
+        #[cfg(feature = "zlib")]
+        let buf = &buf[..std::cmp::min(max_len - RESERVE_SIZE, buf.len())];
```

Comment:
I know this is a temporary hack, but something like this is not sufficient for production. MbedTLS C code may be compiled with zlib enabled even if the Rust bindings are not.

Reply:
I removed this part of code.
After testing with blocking code, test also failed when caller try to write large buffer at one times.

when turn on `zlib` feature, c-mbedtls could not record buffer with size > MBEDTLS_SSL_OUT_CONTENT_LEN (default: 16 * 1024)
https://github.com/fortanix/rust-mbedtls/blob/98331fce9fcd8e9a4e472209aca182329f6de6f2/mbedtls-sys/vendor/library/ssl_msg.c#L646-L653

Disable tests for large buffer when `zlib` is enabled and add negative test case when `zlib` is enabled